src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethod.cs

```diff
@@ -460,7 +460,12 @@ internal RuntimeMethodHandle GetMethodDescriptor()
             bool wrapExceptions = (invokeAttr & BindingFlags.DoNotWrapExceptions) == 0;
 
             StackAllocedArguments stackArgs = default;
-            Span<object?> arguments = CheckArguments(ref stackArgs, parameters, binder, invokeAttr, culture, sig);
+            Span<object?> arguments = default;
+            if (actualCount != 0)
```

Comment:
This check was removed as part of https://github.com/dotnet/runtime/pull/50814 since my perf measurements in that PR showed it wasn't saving any measurable amount of time, so the smaller code was preferred. (See https://github.com/dotnet/runtime/pull/50814/files#r611683957.)

Is this short-circuit now saving time? Is that the 5ns win called out in other rows of the table?

Reply:
When I put the check back into CheckArguments and remove it from the call site, I instead get numbers like this:

| Method |         Toolchain |      Mean | Ratio | Allocated |
|------- |------------------ |----------:|------:|----------:|
|     M0 | \main\CoreRun.exe |  53.37 ns |  1.00 |         - |
|     M0 |   \pr\CoreRun.exe |  55.43 ns |  1.04 |         - |
|        |                   |           |       |           |
|     M1 | \main\CoreRun.exe |  90.49 ns |  1.00 |      32 B |
|     M1 |   \pr\CoreRun.exe |  86.89 ns |  0.96 |      32 B |
|        |                   |           |       |           |
|     R1 | \main\CoreRun.exe |  84.50 ns |  1.00 |      24 B |
|     R1 |   \pr\CoreRun.exe |  83.84 ns |  0.99 |      24 B |
|        |                   |           |       |           |
|  P0Get | \main\CoreRun.exe |  84.79 ns |  1.00 |      24 B |
|  P0Get |   \pr\CoreRun.exe |  86.40 ns |  1.02 |      24 B |
|        |                   |           |       |           |
|  P0Set | \main\CoreRun.exe | 106.57 ns |  1.00 |      56 B |
|  P0Set |   \pr\CoreRun.exe |  94.08 ns |  0.88 |      24 B |
|        |                   |           |       |           |
|  P1Get | \main\CoreRun.exe | 123.81 ns |  1.00 |      24 B |
|  P1Get |   \pr\CoreRun.exe | 123.35 ns |  1.00 |      24 B |
|        |                   |           |       |           |
|  P1Set | \main\CoreRun.exe | 142.47 ns |  1.00 |      64 B |
|  P1Set |   \pr\CoreRun.exe | 140.19 ns |  0.98 |      64 B |